Description: don't pass --version-script options when linking executables,
 only when linking libraries
Author: Julien Cristau <jcristau@debian.org>
Bug-Debian: http://bugs.debian.org/582362

Index: pam/modules/pam_mkhomedir/Makefile.am
===================================================================
--- pam.orig/modules/pam_mkhomedir/Makefile.am
+++ pam/modules/pam_mkhomedir/Makefile.am
@@ -19,14 +19,14 @@
 
 AM_CFLAGS = -I$(top_srcdir)/libpam/include -I$(top_srcdir)/libpamc/include \
 	    -DMKHOMEDIR_HELPER=\"$(sbindir)/mkhomedir_helper\"
-AM_LDFLAGS = -no-undefined -avoid-version -module
-if HAVE_VERSIONING
-  AM_LDFLAGS += -Wl,--version-script=$(srcdir)/../modules.map
-endif
 
 securelib_LTLIBRARIES = pam_mkhomedir.la
 pam_mkhomedir_la_SOURCES = pam_mkhomedir.c
 pam_mkhomedir_la_LIBADD = -L$(top_builddir)/libpam -lpam
+pam_mkhomedir_la_LDFLAGS = -no-undefined -avoid-version -module
+if HAVE_VERSIONING
+  pam_mkhomedir_la_LDFLAGS += -Wl,--version-script=$(srcdir)/../modules.map
+endif
 
 sbin_PROGRAMS = mkhomedir_helper
 mkhomedir_helper_SOURCES = mkhomedir_helper.c
